crates/solvers/src/domain/order.rs

```diff
@@ -0,0 +1,54 @@
+//! The domain object representing a CoW Protocol order.
+
+use crate::domain::eth;
+
+/// A CoW Protocol order in the auction.
+#[derive(Debug, Clone)]
+pub struct Order {
+    pub uid: Uid,
+    pub sell: eth::Asset,
+    pub buy: eth::Asset,
+    pub side: Side,
+    pub class: Class,
+}
+
+/// UID of an order.
+#[derive(Debug, Clone, Copy)]
+pub struct Uid(pub [u8; 56]);
+
+/// The trading side of an order.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Side {
+    /// An order with a fixed buy amount and maximum sell amount.
+    Buy,
+    /// An order with a fixed sell amount and a minimum buy amount.
+    Sell,
+}
+
+/// The order classification.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Class {
+    Market,
+    Limit,
+    Liquidity,
```

Comment:
Might be nice to add some comments here describing the specifics of each type. I guess the most important aspect here in the context of solvers is that `Liquidity` orders should never get surplus.